ContactDetailsApi/V1/Controllers/ContactDetailsController.cs

```diff
@@ -41,5 +44,26 @@ public async Task<IActionResult> GetContactDetailsByTargetId([FromQuery] Contact
 
             return Ok(new GetContactDetailsResponse(contacts));
         }
+
+        /// <summary>
+        /// Soft Deletes contact details for a particular id
+        /// </summary>
+        /// <response code="200">Successfully Operation</response>
+        /// <response code="404">Contact not found for ID requested</response>
+        /// <response code="500">Internal Server Error</response>
+        [ProducesResponseType(typeof(ContactDetailsResponseObject), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpDelete]
+        [LogCall(LogLevel.Information)]
+        public async Task<IActionResult> DeleteContactDetailsByTargetId([FromQuery] ContactQueryParameter queryParam)
+        {
+
+            var contacts = await _deleteContactDetailsByTargetIdUseCase.Execute(queryParam).ConfigureAwait(false);
```

Comment:
_Really_ minor point, but there will only be 1 contact returned so it should be `contact`.